### DISCUSSION TITLE
Ignore formatting for a file/expression?

### DISCUSSION THREAD

**Discussion by nixin72** [kkinnear/zprint#295] [Q&A]
Is there a way to annotate something (akin to `#_{:clj-kondo/ignore ...}`) such that zprint will ignore formatting things within that expression or in a given file?

**Comment by kkinnear** [comment]
Thanks for asking!

I don't use `clj-kondo`, though I certainly know about it.  I spent some time looking over the docs for it, and didn't actually find the `#_{:clj-kondo/ignore ...}` thing, so I'm not much wiser than I was before.   So trying to reason about zprint based on what I know about `clj-kondo` probably isn't going to get us very far, unfortunately.

What zprint __can__ do is accept configuration option maps in comments.    

Thus a comment like this:
```
;!zprint {:format :skip}
```
will cause zprint to skip formatting the next top level (non-comment) element (which is usually a `defn` or `def` expression).   You can skip or change the formatting for a single following expression with the `;!zprint { ... }` directives.  See the [link to the documentation](https://github.com/kkinnear/zprint/blob/main/doc/bang.md) for lots of details and examples.

That is one approach to getting zprint to keep its hands off of something at the top level.  zprint only really knows how to format entire expressions.  I don't know if you are asking for a way to have zprint always *not* format some kind of expression which might appear internally in an outer expression.  zprint won't do that today, though you *could* get zprint to go into `:respect-nl` mode for an internal expression -- which would respect existing new lines while still formatting the code.  That might give you sort of what I kind of imagine you want.  But as you can tell, I don't really __know__ what you want.

It would be very helpful to me if you showed me what zprint does now and told me what you don't like about it and then showed me an example what you would like it to do.  From that I can tell you either how to get that to happen, or I can think about how I could modify zprint to allow that to happen.  

It turns out that there is a "no-format" mode in zprint (specifically to support coloring the code and outputting HTML when using `;!zprint {:format :skip}`), that I suppose could be used internally in an expression to somehow "not format" a piece of an expression.  It probably wouldn't be a lot of work to make that accessible to `:option-fn`, but I don't want to dive into prototyping that until I know better what you really want.

Thanks!
